packages/suite-desktop/src/main/resolveSourcePaths.ts

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
+// SPDX-License-Identifier: MPL-2.0
+
+import fs from "fs";
+import os from "os";
+import path from "path";
+
+import { allowedExtensions } from "./constants";
+
+export function getFilesFromDirectory(arg: string): string[] {
+  try {
+    return fs
+      .readdirSync(arg)
+      .filter((file) =>
+        allowedExtensions.some((extension) => file.toLocaleLowerCase().endsWith(extension)),
+      );
+  } catch (err) {
+    console.error(err);
+  }
+  return [];
+}
+
+export function isPathToDirectory(paths: string[]): boolean {
+  if (paths.length !== 1) {
+    return false;
+  }
+  try {
+    return fs.statSync(paths[0]!).isDirectory();
+  } catch (error) {
+    console.error(error);
+    return false;
+  }
+}
+
+export function resolveSourcePaths(sourceParameter: string | undefined): string[] {
```

Comment:
Using `?` when you call `resolveSourcePaths` is not necessary pass parameter, so you just call `resolveSourcePaths()` instead `resolveSourcePaths(undefinedVarName)`
```suggestion
export function resolveSourcePaths(sourceParameter?: string): string[] {
```